src/modbus-tcp.c

```diff
@@ -401,7 +403,11 @@ static int _modbus_tcp_pi_connect(modbus_t *ctx)
     rc = getaddrinfo(ctx_tcp_pi->node, ctx_tcp_pi->service, &ai_hints, &ai_list);
     if (rc != 0) {
         if (ctx->debug) {
+#ifdef HAVE_GAI_STRERROR
             fprintf(stderr, "Error returned by getaddrinfo: %s\n", gai_strerror(rc));
```

Comment:
why use gai_strerror and not strerror?

Reply:
That's a good question ;) But my MR was only to make it conditional (since in ESP-IDF libs there isn't) not to really change the function used. But in case @stephane could consider that! (and then my change would not be needed indeed)